modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java

```diff
@@ -99,31 +135,100 @@ public void testPostExportTask() throws Exception {
 				"test@able.com", PropsValues.DEFAULT_ADMIN_PASSWORD
 			).apply(
 				() -> {
-					_testPostExportTask("COMPLETED", objectDefinition2);
-					_testPostExportTask("FAILED", objectDefinition1);
+					JSONObject exportTaskJSONObject2 = _testPostExportTask(
+						objectDefinition2, null);
+
+					Assert.assertEquals(
+						"COMPLETED",
+						exportTaskJSONObject2.get("executeStatus"));
+
+					exportTaskJSONObject2 = _testPostExportTask(
+						_objectDefinition1, null);
+
+					Assert.assertEquals(
+						"FAILED", exportTaskJSONObject2.get("executeStatus"));
 				}
 			);
 
-			_objectDefinitionLocalService.deleteObjectDefinition(
-				objectDefinition1);
-			_objectDefinitionLocalService.deleteObjectDefinition(
-				objectDefinition2);
+			_companyLocalService.deleteCompany(
+				companyJSONObject.getLong("companyId"));
+		}
+	}
+
+	@Test
+	public void testPostExportTaskWithFiltering() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.batch.engine.internal." +
+					"BatchEngineExportTaskExecutorImpl",
+				LoggerTestUtil.ERROR)) {
 
-			_companyLocalService.deleteCompany(jsonObject.getLong("companyId"));
+			ObjectEntry objectEntry1 = ObjectEntryTestUtil.addObjectEntry(
+				_objectDefinition1, _OBJECT_FIELD_NAME_TEXT, "TestObject1");
+
+			ObjectEntry objectEntry2 = ObjectEntryTestUtil.addObjectEntry(
+				_objectDefinition1, _OBJECT_FIELD_NAME_TEXT, "TestObject2");
+
+			ObjectEntryTestUtil.addObjectEntry(
+				_objectDefinition1, _OBJECT_FIELD_NAME_TEXT, "Object3");
+
+			String encodedFilterString = URLCodec.encodeURL(
+				StringBundler.concat(
+					"contains(", _OBJECT_FIELD_NAME_TEXT, ", 'Test')"));
+
+			String queryParametersString = "filter=" + encodedFilterString;
+
+			JSONObject jsonObject = _testPostExportTask(
+				_objectDefinition1, queryParametersString);
+
+			Assert.assertEquals(2, jsonObject.getInt("processedItemsCount"));
+
+			InputStream inputStream = HTTPTestUtil.invokeToInputStream(
+				null,
+				StringBundler.concat(
+					"headless-batch-engine/v1.0/export-task",
+					"/by-external-reference-code/",
+					jsonObject.getString("externalReferenceCode"), "/content"),
+				Http.Method.GET);
+
+			ZipInputStream zipInputStream = new ZipInputStream(inputStream);
+
+			zipInputStream.getNextEntry();
+
+			JSONArray responseJSONArray = _jsonFactory.createJSONArray(
+				StringUtil.read(zipInputStream));
```

Comment:
I think it is much more beautiful to use the JSONAssert library. Something like:

JSONAssert.equals(
    JSONUtil.putAll(
      JSONUtil.put(
          "externalReferenceCode", objectEntry1JSONObject.getString("externalReferenceCode")


.....

IT would be just 1 line of code with the expected fields (just the ERC or the text field is enough)